Firestore/core/src/firebase/firestore/local/memory_remote_document_cache.mm

```diff
@@ -35,15 +33,19 @@
 using model::ListenSequenceNumber;
 using model::MaybeDocument;
 using model::MaybeDocumentMap;
+using model::SnapshotVersion;
 using model::OptionalMaybeDocumentMap;
 
 MemoryRemoteDocumentCache::MemoryRemoteDocumentCache(
     MemoryPersistence* persistence) {
   persistence_ = persistence;
 }
 
-void MemoryRemoteDocumentCache::Add(const MaybeDocument& document) {
-  docs_ = docs_.insert(document.key(), document);
+void MemoryRemoteDocumentCache::Add(const MaybeDocument& document,
+                                    const model::SnapshotVersion& read_time) {
+  docs_ = docs_.insert(
+      document.key(),
+      std::pair<MaybeDocument, SnapshotVersion>({document, read_time}));
```

Comment:
Does `std::make_pair(document, read_time)` work here?

C++ has type deduction, just not for class template parameters (yet).

Reply:
It works. Updated PR.